website/static/js/fileViewTreebeard.js

```diff
@@ -0,0 +1,110 @@
+var Fangorn = require('js/fangorn');
+var m = require('mithril');
+
+function FileViewTreebeard(data) {
+
+    // Set item.branch to show the branch of the rendered GitHub file instead of the default branch
+    var addonRootFolders = data['data'][0].children;
+    for (var i=0; i < addonRootFolders.length; i++) {
+        var item = addonRootFolders[i];
+        if (item.provider === 'github' && item.isAddonRoot && window.contextVars.file.extra.branch) {
+            item.branch = window.contextVars.file.extra.branch;
+        }
+    }
+
+    var fangornOpts = {
+    divID: 'grid',
+    filesData: data.data,
+    uploads: false,
+    showFilter: false,
+    title: undefined,
+    hideColumnTitles: true,
+    onload: function(tree) {
+        var tb = this;
+        Fangorn.DefaultOptions.onload.call(tb, tree);
+        $('.osf-panel-header.osf-panel-header-flex').show();
```

Comment:
@sloria, would you mind clarifying what you mean? I'm not sure how to handle this in the view.